Lib/test/test_tempfile.py

```diff
@@ -561,6 +561,23 @@ def test_usable_template(self):
             os.rmdir(d)
 
 
+class TestSanitizeParamsInner(BaseTestCase):
+    """Test the internal function _sanitize_params."""
+
+    def test_throw_exception_on_path_separator_detection(self):
+        with self.assertRaises(ValueError):
+            tempfile.mkstemp(prefix=f"{os.sep}home")
+
+    def test_throw_exception_on_encoded_path_separator_detection(self):
+        with self.assertRaises(ValueError):
+            tempfile.mkstemp(prefix=f"{os.fsencode(os.sep)}home")
```

Comment:
It does not work as you expected. On Posix it generates prefix `"b'/'home"`.

Use `os.fsencode(f"{os.sep}home")`.